### PR TITLE
Feat 2461 - Handle created invitation to event on the backend side

### PR DESCRIPTION
**Pull request by phpgeekfr**
What is left to do:

- [ ] @Mussuranae implement mails in `apps/backend-functions/src/internals/invitations/events.ts`
- [x] @phpgeekfr check `#2461` in code
- [x] @phpgeekfr deploy new functions and test the whole invitation/notifications flow.
- [x] @phpgeekfr implement cron backend function => moved to another issue

### CODE REVIEW

File: libs/notification/src/lib/notification/+state/notification.query.ts [hunk 1]
```diff
@@ -45,11 +45,9 @@ export class NotificationQuery extends QueryEntity<NotificationState, Notificati
           // As Date cannot be used as an index type (key), we format the date into a string.
           const key = isToday(date) ? 'Today'
             : isYesterday(date) ? 'Yesterday'
-            : formatDate(toDate(notification.date), 'MMM dd, yyyy', 'en-US');
-          const information = this.createNotificationInformation(notification);
+              : formatDate(notification.date, 'MM M dd, yyyy', 'en-US');
```
Comment (GrandSchtroumpf) [inline libs/notification/src/lib/notification/+state/notification.query.ts:48]: Why ?

File: libs/notification/src/lib/notification/+state/notification.store.ts [hunk 2]
```diff
@@ -80,44 +85,72 @@ export class NotificationStore extends EntityStore<NotificationState, Notificati
...
+          url: `c/o/marketplace/event/${notification.docId}`
+        };
+      default:
+        return {
+          message: 'Error while displaying notification.'
+        };
+    }
+  }
+
+  private notificationSubject(notification) {
```
Comment (GrandSchtroumpf) [inline libs/notification/src/lib/notification/+state/notification.store.ts:144]: type the parameter

File: apps/backend-functions/src/admin.ts [hunk 3]
```diff
@@ -27,13 +26,16 @@ import * as backup from './backup';
...
 
 /**
- * Handles firestore update on request to application access,
+ * Handles firestore update on request to application access
+ * @TODO (#2539)
+ * This method is currently unused but we keep it to future uses.
+ * It sends and email to admin to accept or reject the request
  */
-export async function onRequestAccessToAppWrite(
+/*export async function onRequestAccessToAppWrite(
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply apps/backend-functions/src/admin.ts:38]: Vincent told me that we are going to reuse that.

File: apps/backend-functions/src/internals/invitations/utils.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { InvitationDocument, InvitationOrUndefined } from "../../data/types";
+
+/** Checks if an invitation just got accepted. */
+export function wasAccepted(before: InvitationDocument, after: InvitationDocument) {
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply apps/backend-functions/src/internals/invitations/utils.ts:4]: Don't think so. We don't have before and after state on front

File: libs/notification/src/lib/notification/+state/notification.query.ts (hunk 1 above)
Comment (phpgeekfr) [reply libs/notification/src/lib/notification/+state/notification.query.ts:48]: notification.date is already a date.

Comment (GrandSchtroumpf) [reply libs/notification/src/lib/notification/+state/notification.query.ts:48]: ok, format the tabs as before please :)

**Pull request merged**